### PR TITLE
SwiftRemoteMirror: Add reflection metadata version query API

### PR DESCRIPTION
**Pull request by bitjammer**
Add a major/minor version API to the SwiftRemoteMirror library. These numbers will correspond to some constants emitted in Swift binaries.

rdar://problem/27251582

### CODE REVIEW

**Comment by bitjammer** [comment]
@slavapestov Can you review? I can follow up with the IRGen change.

File: include/swift/Reflection/Records.h [hunk 1]
```diff
@@ -19,6 +19,9 @@
 
 #include "swift/Basic/RelativePointer.h"
 
+const uint16_t SWIFT_REFLECTION_METADATA_VERSION_MAJOR = 1;
```
Comment (slavapestov) [inline include/swift/Reflection/Records.h]: Perhaps we should just have a single version number. Do we expect to handle major number mismatches differently from minor?

Comment (bitjammer) [inline include/swift/Reflection/Records.h]: I was mostly just adding it for flexibility. It could probably be one number.

**Comment by bitjammer** [comment]
@swift-ci Please smoke test

**Comment by bitjammer** [comment]
@swift-ci Please smoke test

**Comment by bitjammer** [comment]
@swift-ci Please smoke test

**Pull request merged**